### PR TITLE
Add recipe for pygad

### PR DESCRIPTION
**Pull request by traversaro**
The recipe was created via [grayskull](https://www.marcelotrevisani.com/grayskull).

cc @GiulioRomualdi @carlottasartore 

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by traversaro** [comment]
It took me a while to realize that the content of the source package on pypi does not match at all the content of the https://github.com/ahmedfgad/GeneticAlgorithmPython. Instead, the code is distributed among several tag-less repos, see https://github.com/ahmedfgad/GeneticAlgorithmPython/tree/2.18.1#pygad-source-code . For this reason, it is important to always use the source code from pypi, even if this may complicate the application of patches.

**Comment by traversaro** [comment]
> For this reason, it is important to always use the source code from pypi, even if this may complicate the application of patches.

For example, in this case to adapt https://github.com/ahmedfgad/GeneticAlgorithmPython/pull/152 to the PyPI source archive we need to change the location of the `pygad.py` file to patch, see https://github.com/conda-forge/staged-recipes/pull/21719/commits/dcecc12bd3c5082ad420a10b38b847636e3cc46a .

File: recipes/pygad/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    - python >=3.7
+    - numpy
+    - matplotlib-base
+
+test:
+  imports:
+    - pygad
+  commands:
+    - pip list --verbose
+    - python -c "import pygad"
```
Comment (dopplershift) [inline recipes/pygad/meta.yaml:39]: This is the same as the line above under `imports`:
```suggestion
```

Comment (traversaro) [reply recipes/pygad/meta.yaml]: Thanks! Indeed it was a debugging leftover.

**Comment by traversaro** [comment]
@conda-forge/help-python The recipe is ready for review.

**Review by dopplershift — APPROVED** [review]

**Pull request merged**